contracts/InterchainTokenService.sol

```diff
@@ -707,8 +714,10 @@ contract InterchainTokenService is
     function _checkPayloadAgainstGatewayData(bytes memory payload, string calldata tokenSymbol, uint256 amount) internal view {
         (, bytes32 tokenId, , , uint256 amountInPayload) = abi.decode(payload, (uint256, bytes32, uint256, uint256, uint256));
 
-        if (validTokenAddress(tokenId) != gatewayAddress.tokenAddresses(tokenSymbol) || amount != amountInPayload)
-            revert InvalidGatewayTokenTransfer(tokenId, payload, tokenSymbol, amount);
+        if (
+            validTokenAddress(tokenId) != IAxelarGMPGatewayWithToken(address(gateway)).tokenAddresses(tokenSymbol) ||
```

Comment:
```suggestion
        if (
            validTokenAddress(tokenId) != gatewayWithToken().tokenAddresses(tokenSymbol) ||
```
create an internal gatewayWithToken method since this needs to be done in more than one place (if it was a single one, then i would have recommended to create a variable for it)

Reply:
Yeah, I agree with your approach. Updated.